### DISCUSSION TITLE
False classification of Python/Brainfuck repo as HyPhy

### DISCUSSION THREAD

**Discussion by obfuscatedgenerated** [github-linguist/linguist#5592] [Classification] [answered]
This repo: [obfuscatedgenerated/brainfComp](https://github.com/obfuscatedgenerated/brainfComp) is being shown to have 0.2% HyPhy content. Furthermore, clicking this percentage shows no results, as proven [here](https://github.com/obfuscatedgenerated/brainfComp/search?l=hyphy).

**Comment by lildude** [comment, marked as answer]
Linguist is detecting the `bell.bf` file as HyPhy:

```console
$ github-linguist brainfComp --breakdown
70.91%  2803       Python
28.89%  1142       Brainfuck
0.20%   8          HyPhy

HyPhy:
bell.bf

Python:
brainfComp.py
brainfFromFile.py
brainfScribe.py

Brainfuck:
divide.bf
helloworld.bf


$
```

This is happening because HyPhy and Brainfuck (and Beef) share the same extension and we rely on the samples to train the classifier to try and distinguish the difference. In the case of the `bell.bf` file, it's a very small file so the classifier will struggle to get it right so you will need to implement an override to force the correct language.

As for getting no results when you click the language in the sidebar: this will something going awry with the Search functionality and has nothing to do with Linguist.

**Reply by obfuscatedgenerated** [reply]
Is the override something I can do from .gitattributes?

`*.bf linguist-language=Brainfuck`

**Reply by obfuscatedgenerated** [reply]
That worked, thanks!
